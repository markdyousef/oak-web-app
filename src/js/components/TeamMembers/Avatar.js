// @flow
import React, { PropTypes } from 'react';

import css from './Member';

// import placeholderImg
const placeholderImg = null;

const Avatar = (img: String) => (
    <div className={css.avatar}>
        <img src={img} alt="profile" />
    </div>
);

Avatar.propTypes = {
    img: PropTypes.string
};

Avatar.defaultProps = {
    img: placeholderImg
};

export default Avatar;
