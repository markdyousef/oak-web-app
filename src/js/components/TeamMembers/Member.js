// @flow
import React, { PropTypes } from 'react';
import { Link } from 'react-router';

import css from './TeamMembers.css';

const Member = ({ member }: Object) => (
    <Link
        className={css.member}
        to={`/user/${member}`}
    >
        <div />
        <h4>{member.name}</h4>
    </Link>
);

Member.propTypes = {
    member: PropTypes.shape({
        name: PropTypes.string.isRequired
    })
};

Member.defaultProps = {
    member: {}
};

export default Member;
