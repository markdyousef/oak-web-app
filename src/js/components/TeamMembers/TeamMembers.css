.container {
    width: 100%;
    display: flex;
    height: 300px;
}
