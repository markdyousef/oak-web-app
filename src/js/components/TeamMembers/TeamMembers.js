import React, { Component, PropTypes } from 'react';

import css from './TeamMembers.css';

class TeamMembers extends Component {
    static propTypes = {
    }
    constructor() {
        super();
        this.state = {};
    }
    render() {
        return (
            <div className={css.container}>
                <h1>Members</h1>
            </div>
        );
    }
}

export default TeamMembers;
