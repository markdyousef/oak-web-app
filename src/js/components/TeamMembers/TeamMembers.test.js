// @flow
import React from 'react';
// import renderer from 'react-test-renderer';
import { shallow } from 'enzyme';
import TeamMembers from './TeamMembers';

test('renders TeamMembers', () => {
    const props = {
    };
    const shallowComponent = shallow(
        <TeamMembers {...props} />
    );
    expect(shallowComponent).toMatchSnapshot();
});
