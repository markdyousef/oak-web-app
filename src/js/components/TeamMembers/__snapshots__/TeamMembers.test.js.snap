exports[`test renders TeamMembers 1`] = `
ShallowWrapper {
  "complexSelector": ComplexSelector {
    "buildPredicate": [Function],
    "childrenOfNode": [Function],
    "findWhereUnwrapped": [Function],
  },
  "length": 1,
  "node": <div
    className={undefined}>
    <h1>
        Members
    </h1>
</div>,
  "nodes": Array [
    <div
      className={undefined}>
      <h1>
            Members
      </h1>
</div>,
  ],
  "options": Object {},
  "renderer": ReactShallowRenderer {
    "_instance": ShallowComponentWrapper {
      "_calledComponentWillUnmount": false,
      "_compositeType": 0,
      "_context": Object {},
      "_currentElement": <TeamMembers />,
      "_debugID": 1,
      "_hostContainerInfo": null,
      "_hostParent": null,
      "_instance": TeamMembers {
        "_reactInternalInstance": [Circular],
        "context": Object {},
        "props": Object {},
        "refs": Object {},
        "state": Object {},
        "updater": Object {
          "enqueueCallback": [Function],
          "enqueueCallbackInternal": [Function],
          "enqueueElementInternal": [Function],
          "enqueueForceUpdate": [Function],
          "enqueueReplaceState": [Function],
          "enqueueSetState": [Function],
          "isMounted": [Function],
          "validateCallback": [Function],
        },
      },
      "_mountOrder": 1,
      "_pendingCallbacks": null,
      "_pendingElement": null,
      "_pendingForceUpdate": false,
      "_pendingReplaceState": false,
      "_pendingStateQueue": null,
      "_renderedComponent": NoopInternalComponent {
        "_currentElement": <div
          className={undefined}>
          <h1>
                    Members
          </h1>
</div>,
        "_debugID": 2,
        "_renderedOutput": <div
          className={undefined}>
          <h1>
                    Members
          </h1>
</div>,
      },
      "_renderedNodeType": 0,
      "_rootNodeID": 0,
      "_topLevelWrapper": null,
      "_updateBatchNumber": null,
      "_warnedAboutRefsInRender": false,
    },
    "getRenderOutput": [Function],
    "render": [Function],
  },
  "root": [Circular],
  "unrendered": <TeamMembers />,
}
`;
